velox/connectors/hive/iceberg/tests/IcebergPartitionIdGeneratorTest.cpp

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/iceberg/IcebergPartitionIdGenerator.h"
+#include "velox/common/encode/Base64.h"
+#include "velox/connectors/hive/iceberg/Transform.h"
+#include "velox/connectors/hive/iceberg/tests/IcebergTestBase.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+
+using namespace facebook::velox;
+
+namespace {
+
+class IcebergPartitionIdGeneratorTest : public test::IcebergTestBase {
+ protected:
+  std::vector<TransformPtr> createColumnTransforms(
+      const std::vector<std::string>& columnNames,
+      const std::vector<TypePtr>& types,
+      const std::vector<TransformType>& transformTypes,
+      const std::vector<std::optional<int32_t>>& parameters = {}) {
+    std::vector<IcebergPartitionSpec::Field> fields;
+    fields.reserve(columnNames.size());
+
+    for (auto i = 0; i < columnNames.size(); ++i) {
+      auto parameter = parameters.size() > i ? parameters.at(i) : std::nullopt;
+      fields.emplace_back(
+          IcebergPartitionSpec::Field{
+              columnNames[i], types[i], transformTypes[i], parameter});
+    }
+
+    spec_ = std::make_shared<const IcebergPartitionSpec>(0, fields);
```

Comment:
It is an anti-pattern to have helper methods with side effects. Is this necessary?